go/vt/vttablet/tabletserver/tabletenv/config.go

```diff
@@ -206,6 +210,7 @@ var DefaultQsConfig = TabletConfig{
 	QueryPoolTimeout:        0,
 	TxPoolTimeout:           1,
 	IdleTimeout:             30 * 60,
+	QueryPoolWaiterCap:      0,
```

Comment:
Given that this is a form of OOM protection, I think we should set a reasonably high value as default instead, like 50,000. Also need to initialize TxPoolWaiterCap.

This will also allow us to remove `0` as special case: people must set a value for this.